Domain/Repositories/BookingRepository.cs

```diff
@@ -0,0 +1,155 @@
+                return;
+            }
+            _bookings.Remove(booking);
+            Console.WriteLine($"Booking for {bookingId} canceled successfully.");
+            SaveBookings();
+        }
+
+        public List<Booking> GetAllBookings()
+        {
+            return _bookings;
+        }
+
+        public Booking GetLastBooking()
+        {
+            return _bookings.Last();
+        }
+
+        public void UpdateBooking(Booking updatedBooking)
+        {
+            var existingBooking = GetBookingById(updatedBooking.Id);
+            if (existingBooking != null)
+            {
+                existingBooking.FlightId = updatedBooking.FlightId;
+                existingBooking.SeatClass = updatedBooking.SeatClass;
+                SaveBookings();
+            }
+        }
+
+        private void LoadBookings()
+        {
+            var lines = File.ReadAllLines(_filePath);
+            foreach (var line in lines.Skip(1))
+            {
+                var parts = line.Split(',');
+                if (parts.Length == 7)
+                {
+                    try
+                    {
+                        var _passenger = new Passenger
+                        {
+                            Id = int.Parse(parts[2]),
+                            Name = parts[3],
+                            Email = parts[4]
+                        };
+
+
+
+                        var booking = new Booking
+                        {
+                            Id = int.Parse(parts[0]),
+                            FlightId = int.Parse(parts[1]),
+                            SeatClass = parts[5],
+                            Passenger = _passenger,
+                            BookDate = DateTime.Parse(parts[6])
+                        };
+                            
+                        _bookings.Add(booking);
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine($"Error in loading the trip: {ex.Message}");
+                    }
+                }
+            }
+        }
+
+        public List<(Booking Booking, Flight Flight)> FilteredBookings(
+               List<Flight> filteredFlights, double? price, string seatClass, string passenger)
+        {
+            var filteredBookings = _bookings
+                .Where(b =>
+                    filteredFlights.Any(f => f.Id == b.FlightId) &&
+                    (price == null || _flightsService.GetPriceByClass(
+                        filteredFlights.FirstOrDefault(f => f.Id == b.FlightId), seatClass) <= price.Value) &&
+                    (string.IsNullOrEmpty(passenger) || b.Passenger.Name.Equals(passenger, StringComparison.OrdinalIgnoreCase)) &&
+                    (string.IsNullOrEmpty(seatClass) || b.SeatClass.Equals(seatClass, StringComparison.OrdinalIgnoreCase)))
+                .Select(b => (Booking: b, Flight: filteredFlights.FirstOrDefault(f => f.Id == b.FlightId)))  
```

Comment:
you can save the flight in the booking model itself, instead of saving there the flight id